zeppelin-zengine/src/test/java/org/apache/zeppelin/interpreter/SparkDownloadUtils.java

```diff
@@ -0,0 +1,91 @@
+package org.apache.zeppelin.interpreter;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+public class SparkDownloadUtils {
+  private static Logger LOGGER = LoggerFactory.getLogger(SparkDownloadUtils.class);
+
+  private static String downloadFolder = System.getProperty("user.home") + "/.cache/spark";
```

Comment:
I don't know how much we are supporting Windows nowadays, but we would need to be a different subpath

Reply:
I am afraid we need to change a lot for windows support besides this, zeppelin doesn't officially support windows IIUC, even we add `.appveyor.yml`, there's no build there actually.

Reply:
well, we used to get a fair number of questions related to running zeppelin on windows in user@ or dev@, I think we should raise this before the next release with these changes...

Reply:
make sense